Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -261,10 +262,12 @@ public class PBXProjGenerator {
                     if dependencyTarget.type.isExtension {
                         // embed app extension
                         extensions.append(embedFile.reference)
+                    } else if dependencyTarget.type.isFramework {
```

Comment:
We'd need a bit more check here since the extension is not the only one determining if the framework needs to be copied. Only the dynamic ones need to. You can use [lipo](https://pewpewthespells.com/blog/convert_static_to_dynamic.html) for that.

Reply:
Yea, I thought about that case, we actually have that case and I'm currently setting `embed: False` on it since we don't currently have a reference to the actual binary.

Reply:
We can also use `file` which might be easier than `lipo` for this

Reply:
I thought this was at least an improvement since that's now the only case we're not handling correctly after this and my other PRs

Reply:
I agree this could be the second iteration to make it more concrete 👍. I didn't know it was possible to get that info using `file`.